streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -800,6 +807,9 @@ long runOnce(final long recordsProcessedBeforeCommit) {
         if (records != null && !records.isEmpty() && taskManager.hasActiveRunningTasks()) {
             streamsMetrics.pollTimeSensor.record(computeLatency(), timerStartedMs);
             addRecordsToTasks(records);
+            if (taskManager.versionProbingFlag) {
```

Comment:
I am not 100% sure if we need this or not. When we get am empty assignment, I would actually assume that `poll()` returns no records, and thus we would never hit this ?

Reply:
Yes that is correct, we always cleanup the internal buffer during rebalance.

Reply:
Ack. Than I will remove this completely. Don't think we need to throw.

I just put it in to get feedback on the PR -- otherwise, it's hard to explain my thoughts :)